src/server/origin/address/controller.test.js

```diff
@@ -257,19 +259,25 @@ describe('#originAddressController', () => {
     )
 
     expect(statusCode).toBe(statusCodes.redirect) // Assuming a redirect on success
-    expect(headers.location).toBe('/origin-address') // Check for redirect response
+    expect(headers.location).toBe('/origin/summary') // Check for redirect response
   })
 
   testCsrfProtectedGet(() => server, {
     method: 'GET',
-    url: '/origin-address'
+    url: '/origin/address'
   })
 
   testCsrfProtectedPost(() => server, {
     method: 'POST',
-    url: '/origin-address',
+    url: '/origin/address',
     payload: {
-      originAddress: 'on'
+      originAddress: {
```

Comment:
I know it isn't relevant here, but the payload for POST will always be an object with: 

```
{ [key: string]: string }
```

because that's the form data payload that the framework's expecting.

Reply:
this locks it down to `{{
 *  addressLine1: string;
 *  addressLine2 ?: string;
 *  addressTown: string;
 *  addressCounty ?: string;
 *  addressPostcode: string;
 * }}` though meaning that we know what keys we are fetching - if we accept the "any key string" principle, then there is no need to have typescript as it could be anything, we are saying its not just anything

Reply:
Sorry, this was poorly phrased. 

the payload isn't nested within `originAddress` - it's just fields beneath that.  e.g.

```javascript
{ 
    addressLine1: 'Starfleet Headquarters',
    addressLine2: '24-593 Federation Drive',
    addressTown: 'San Francisco',
    addressCounty: 'San Francisco',
    addressPostcode: 'RG24 8RR'
}
```
rather than what it currently has:

```javascript
{ 
  originAddress: { 
    addressLine1: 'Starfleet Headquarters',
    addressLine2: '24-593 Federation Drive',
    addressTown: 'San Francisco',
    addressCounty: 'San Francisco',
    addressPostcode: 'RG24 8RR'
  }
}
```

(forgive my crimes against indentation, the comment editor isn't great at that.